packages/language-service/src/features/code-fix.ts

```diff
@@ -0,0 +1,114 @@
+import type { LanguageServiceWorker } from "../private.js";
+import {
+  type ProtocolPosition,
+  type ProtocolRange,
+} from "../utils/position.js";
+import { Range } from "@knuckles/location";
+
+export interface DiagnosticIdentifier {
+  code: string;
+  range: ProtocolRange;
+}
+
+export interface CodeFixesParams {
+  fileName: string;
+  position: ProtocolPosition;
+  diagnostics?: DiagnosticIdentifier[];
+}
+
+export interface CodeFix {
+  label: string;
+  edits: CodeFixEdit[];
+  diagnostic?: DiagnosticIdentifier;
+}
+
+export type CodeFixEdit =
+  | {
+      type: "create-file";
+      fileName: string;
+    }
+  | {
+      type: "delete-file";
+      fileName: string;
+    }
+  | {
+      type: "rename-file";
+      oldFileName: string;
+      newFileName: string;
+    }
+  | {
+      type: "delete";
+      fileName: string;
+      range: ProtocolRange;
+    }
+  | {
+      type: "replace";
+      fileName: string;
+      range: ProtocolRange;
+      text: string;
+    }
+  | {
+      type: "insert";
+      fileName: string;
+      position: ProtocolPosition;
+      text: string;
+    };
+
+export type CodeFixes = CodeFix[];
+
+export default async function getCodeFixes(
```

Comment:
This language service feature should be named "code actions". Rename all instances of "code fix" to "code action"